Homework 27

TetianaZaikina opened the pull request:
Failed. Both with this homework and the course.
Thank you for your help!

TetianaZaikina committed:
Update LoginTests.java

TetianaZaikina committed:
Merge branch 'main' of https://github.com/TetianaZaikina/A49-QA-Automation-Framework

TetianaZaikina committed:
homework-27

Failed. Test steps not received.
I am confused with the lesson info and the given code.
Anyway,
Thank you for your help!

Berichicko commented:
You got this problem, please correct it. 

![image](https://github.com/Test-Pro-Bootcamp/A49-QA-Automation-Framework/assets/104761482/38343cda-604d-4cf7-9423-55d2a7b5b404)

HINT

I have corrected all the errors, please check [LINK ](https://drive.google.com/file/d/1UqSG2eMu9OYDWDeuJzGCdj6T0uSxxuoo/view?usp=sharing) and download

TetianaZaikina committed:
homework-27

-corrected accordingly to the notes
-implemented 5 steps with corrections in the LoginStepDefinition

still do not see what's going wrong

TetianaZaikina commented:
Thank you for your patience and time. I have applied the corrections.  But it still does not work.

Berichicko commented:
You got this problem, please correct it. 

![image](https://github.com/Test-Pro-Bootcamp/A49-QA-Automation-Framework/assets/104761482/1e126f95-7d67-419d-923f-70687efa5596)

HINT

1) Create the **class HomePage**

![image](https://github.com/Test-Pro-Bootcamp/A49-QA-Automation-Framework/assets/104761482/0cffdbcb-1d32-46ee-80fe-92b6fc6ed17c)


```
package pageObjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class HomePage extends BasePage{
    @FindBy(css = ".avatar")
    private WebElement userAvatarIcon;
    // This annotation is used to locate the user avatar icon on the home page using CSS selector.

    public HomePage(WebDriver givenDriver) {
        super(givenDriver);
        // The constructor calls the constructor of the base class (BasePage) using the given WebDriver.
    }

    public boolean isAvatarDisplayed() {
        return findElement(userAvatarIcon).isDisplayed();
        // This method returns the userAvatarIcon WebElement, representing the user avatar element on the home page.
    }
}
```

2) Project structure: classes should be in appropriate folders

![image](https://github.com/Test-Pro-Bootcamp/A49-QA-Automation-Framework/assets/104761482/1ecb98b9-a6bd-4105-8125-ee6b05005260)

3) 

```
import io.cucumber.java.After;
import io.cucumber.java.Before;
import io.cucumber.java.en.And;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import pageObjects.HomePage;
import pageObjects.LoginPage;

import java.time.Duration;

public class LoginStepDefinitions {
    WebDriver driver;
    WebDriverWait wait;

    @Before
    public void openBrowser() {
        WebDriverManager.chromedriver().setup();
        ChromeOptions options = new ChromeOptions();
        options.addArguments("--disable-notifications");
        options.addArguments("--remote-allow-origins=*");
        driver = new ChromeDriver(options);
        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
    }

    @After
    public void closeBrowser() {
        driver.quit();
    }

    @Given("I open Login page")
    public void openLoginPage() {
        driver.get("https://qa.koel.app/");
    }

    @When("I enter email {string}")
    public void i_enter_email(String email) {
        LoginPage loginPage = new LoginPage(driver);
        loginPage.provideEmail(email);
    }

    @And("I enter password {string}")
    public void i_enter_password(String password) {
        LoginPage loginPage = new LoginPage(driver);
        loginPage.providePassword(password);
    }

    @And("I submit")
    public void clickSubmit() {
        LoginPage loginPage = new LoginPage(driver);
        loginPage.clickSubmit();
    }

    @Then("I am logged in")
    public void userIsLoggedIn() {
        HomePage homePage = new HomePage(driver);
        Assert.assertTrue(homePage.isAvatarDisplayed());
    }
}

```
